framework/src/modules/token/commands/transfer.ts

```diff
@@ -11,38 +11,96 @@
  *
  * Removal or modification of this copyright notice is prohibited.
  */
+
+import * as cryptography from '@liskhq/lisk-cryptography';
 import { validator } from '@liskhq/lisk-validator';
+import { dataStructures } from '@liskhq/lisk-utils';
 import { BaseCommand } from '../../base_command';
 import {
 	CommandExecuteContext,
 	CommandVerifyContext,
+	MethodContext,
 	VerificationResult,
 	VerifyStatus,
 } from '../../../state_machine';
 import { TokenMethod } from '../method';
 import { transferParamsSchema } from '../schemas';
+import { UserStore } from '../stores/user';
+import { InitializeUserAccountEvent } from '../events/initialize_user_account';
+import { TokenID } from '../types';
+import { TransferEvent } from '../events/transfer';
 
 interface Params {
-	tokenID: Buffer;
+	tokenID: TokenID;
 	amount: bigint;
 	recipientAddress: Buffer;
 	data: string;
+	accountInitializationFee: bigint;
 }
 
 export class TransferCommand extends BaseCommand {
 	public schema = transferParamsSchema;
 	private _method!: TokenMethod;
+	private _accountInitializationFee!: bigint;
+	private _feeTokenID!: TokenID;
 
-	public init(args: { method: TokenMethod }) {
+	public init(args: {
+		method: TokenMethod;
+		feeTokenID: TokenID;
+		accountInitializationFee: bigint;
+	}) {
 		this._method = args.method;
+		this._feeTokenID = args.feeTokenID;
+		this._accountInitializationFee = args.accountInitializationFee;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
 	public async verify(context: CommandVerifyContext<Params>): Promise<VerificationResult> {
 		const { params } = context;
 
 		try {
 			validator.validate(transferParamsSchema, params);
+
+			if (context.transaction.senderAddress.equals(params.recipientAddress)) {
```

Comment:
It should be allowed to send transfer to  itself, also i think this is not defined in LIP?